src/messaging/ReliableMessageMgr.cpp

```diff
@@ -350,34 +350,33 @@ bool ReliableMessageMgr::CheckAndRemRetransTable(ReliableMessageContext * rc, ui
 
 CHIP_ERROR ReliableMessageMgr::SendFromRetransTable(RetransTableEntry * entry)
 {
-    CHIP_ERROR err              = CHIP_NO_ERROR;
     ReliableMessageContext * rc = entry->rc;
-    uint32_t msgId              = 0; // Not actually used unless we reach the
-                                     // line that initializes it properly.
-
-    VerifyOrReturnError(rc != nullptr, err);
-
-    // Now that we know this is a valid entry, grab the message id from the
-    // retained buffer.  We need to do that now, because we're about to hand it
-    // over to someone else, and on failure it will no longer be available.
-    msgId = entry->retainedBuf.GetMsgId();
+    if (rc == nullptr)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
```

Comment:
This is changing behavior from returning `CHIP_NO_ERROR` to returning `CHIP_ERROR_INCORRECT_STATE`.  Is that on purpose?  It might be better to do the behavior change as a separate PR...